src/uu/pmap/src/pmap.rs

```diff
@@ -88,7 +89,47 @@ fn parse_maps(pid: &str) -> Result<u64, Error> {
         total += map_line.size_in_kb;
     }
 
-    Ok(total)
+    println!(" total {total:>16}K");
+
+    Ok(())
+}
+
+fn output_device_format(pid: &str) -> Result<(), Error> {
```

Comment:
Seems that there are some duplication here.

What about doing something like:
```
fn process_maps<F>(pid: &str, mut process_line: F) -> Result<(), Error>
where
    F: FnMut(&MapLine),
{
    let path = format!("/proc/{pid}/maps");
    let contents = fs::read_to_string(path)?;
    
    for line in contents.lines() {
        let map_line = parse_map_line(line);
        process_line(&map_line);
    }
    
    Ok(())
}


```

and 
```
fn output_default_format(pid: &str) -> Result<(), Error> {
    let mut total = 0;
    process_maps(pid, |map_line| {
        total += map_line.size_in_kb;
    })?;
    println!(" total {total:>16}K");
    Ok(())
}
```

Reply:
Good point, thanks!